modules/@shopify/checkout-sheet-kit/ios/AcceleratedCheckoutButtons.swift

```diff
@@ -0,0 +1,418 @@
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import Foundation
+import PassKit
+import React
+import ShopifyCheckoutSheetKit
+import SwiftUI
+import UIKit
+
+// MARK: - AcceleratedCheckout Components
+
+@available(iOS 16.0, *)
+class AcceleratedCheckoutConfiguration {
+    static let shared = AcceleratedCheckoutConfiguration()
+    var configuration: ShopifyAcceleratedCheckouts.Configuration?
+    var applePayConfiguration: ShopifyAcceleratedCheckouts.ApplePayConfiguration?
+
+    // TODO: Replace this with public defaults from the kit when available
+    var defaultWallets: [Wallet] = [.applePay, .shopPay]
+
+    var available: Bool {
+        if #available(iOS 16.0, *) {
+            return configuration != nil
+        } else {
+            return false
+        }
+    }
+
+    var applePayAvailable: Bool {
+        if #available(iOS 16.0, *) {
+            return applePayConfiguration != nil
+        } else {
+            return false
+        }
+    }
+}
+
+@objc(RCTAcceleratedCheckoutButtonsManager)
+class RCTAcceleratedCheckoutButtonsManager: RCTViewManager {
+    override func view() -> UIView! {
+        if #available(iOS 16.0, *) {
+            return RCTAcceleratedCheckoutButtonsView()
+        }
+
+        // Return an empty view for iOS < 17.0 (silent fallback)
+        return UIView()
+    }
+
+    override static func requiresMainQueueSetup() -> Bool {
+        return true
+    }
+
+    override func constantsToExport() -> [AnyHashable: Any]! {
+        return [:]
+    }
+}
+
+@available(iOS 16.0, *)
+class RCTAcceleratedCheckoutButtonsView: UIView {
+    private var hostingController: UIHostingController<AnyView>?
+    private var configuration: ShopifyAcceleratedCheckouts.Configuration?
+    private weak var parentViewController: UIViewController?
+
+    @objc var onSizeChange: RCTDirectEventBlock?
+
+    // MARK: - Props
+
+    /// Note that prop values are intentionally nil so that the kit defaults are used
+
+    /**
+     * Accepts either { cartId } or { variantId, quantity }.
+     */
+    @objc var checkoutIdentifier: NSDictionary? {
+        didSet {
+            updateView()
+        }
+    }
+
+    @objc var cornerRadius: NSNumber? {
+        didSet {
+            updateView()
+        }
+    }
+
+    @objc var wallets: [String]? {
+        didSet {
+            invalidateIntrinsicContentSize()
+            setNeedsLayout()
+            updateView()
+        }
+    }
+
+    @objc var applePayLabel: String? {
+        didSet {
+            updateView()
+        }
+    }
+
+    @objc var onPress: RCTBubblingEventBlock?
```

Comment:
TODO: I don't think this is actually needed now that the height issues are fixed